hack/tools/release/weekly/main.go

```diff
@@ -224,14 +203,82 @@ func trimTitle(title string) string {
 	return strings.TrimSpace(title)
 }
 
-type commit struct {
-	merge string
-	body  string
-}
-
 func formatMerge(line, prNumber string) string {
 	if prNumber == "" {
 		return line
 	}
-	return fmt.Sprintf("%s (%s)", line, prNumber)
+	return fmt.Sprintf("%s (#%s)", line, prNumber)
+}
+
+// Parse the time from string to time.Time format
+func parseTime(date string) (time.Time, error) {
+	datetime, err := time.Parse(timeLayout, date)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return datetime, nil
+}
+
+func getMergedPullRequests(client *github.Client, owner string, repo string, branch string, since time.Time, until time.Time) ([]*github.PullRequest, error) {
+	var allPullRequests []*github.PullRequest
+
+	// List up to 500 most recent pull requests to not overwhelm github
+	// This is a best effort alternative over paginating through all PRs
+	// as there are no filters for the time via sdk.
+	listOptions := &github.ListOptions{PerPage: 500}
+
+	pullRequests, _, err := client.PullRequests.List(context.Background(), owner, repo, &github.PullRequestListOptions{
+		State:       "closed",
+		Sort:        "updated",
+		Direction:   "desc",
+		ListOptions: *listOptions,
+		Base:        branch,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Include PRs until EOD
+	until = until.Add(time.Hour * 24)
+	for _, pr := range pullRequests {
+		if pr.MergedAt != nil && pr.MergedAt.After(since) && pr.MergedAt.Before(until) {
+			allPullRequests = append(allPullRequests, pr)
+		}
+	}
+
+	return allPullRequests, nil
+}
+
+// Checks if the branch exists
+func isValidBranch(branchName string, owner string, repo string, client *github.Client) (bool, error) {
+	branches, _, err := client.Repositories.ListBranches(context.Background(), owner, repo, nil)
+	if err != nil {
+		return false, err
+	}
+	for _, branch := range branches {
+		if branchName != branch.GetName() {
+			continue
+		}
+		return true, nil
```

Comment:
```suggestion
		if branchName == branch.GetName() {
			return true, nil
		}
```
super nit to reduce lines of code